Failure is unrelated, and tracked in https://github.com/moby/moby/issues/44128 (still quite flaky that one)

```
=== FAIL: amd64.integration-cli TestDockerCLIRunSuite/TestRunAttachDetachFromInvalidFlag (0.41s)
    docker_cli_run_unix_test.go:232: read |0: file already closed
    --- FAIL: TestDockerCLIRunSuite/TestRunAttachDetachFromInvalidFlag (0.41s)
```

--- a/daemon/containerd/image_snapshot.go
+++ b/daemon/containerd/image_snapshot.go
@@ -3,6 +3,7 @@ package containerd
 import (
 	"context"
 
+	"github.com/containerd/containerd"
 	containerdimages "github.com/containerd/containerd/images"
 	"github.com/containerd/containerd/leases"
 	"github.com/containerd/containerd/platforms"
@@ -12,7 +13,7 @@ import (
 
 // PrepareSnapshot prepares a snapshot from a parent image for a container
 func (i *ImageService) PrepareSnapshot(ctx context.Context, id string, parentImage string, platform *ocispec.Platform) error {
-	desc, err := i.resolveDescriptor(ctx, parentImage)
+	img, err := i.resolveImage(ctx, parentImage)
 	if err != nil {
 		return err
 	}
@@ -24,7 +25,19 @@ func (i *ImageService) PrepareSnapshot(ctx context.Context, id string, parentIma
 		matcher = platforms.Only(*platform)
 	}
 
-	desc, err = containerdimages.Config(ctx, cs, desc, matcher)
+	platformImg := containerd.NewImageWithPlatform(i.client, img, matcher)
+	unpacked, err := platformImg.IsUnpacked(ctx, i.snapshotter)
+	if err != nil {
+		return err
+	}
+
+	if !unpacked {
+		if err := platformImg.Unpack(ctx, i.snapshotter); err != nil {
+			return err
+		}
+	}
+
+	desc, err := containerdimages.Config(ctx, cs, img.Target, matcher)
 	if err != nil {
 		return err
 	}

